als-server/shared/src/main/scala/org/mulesoft/als/server/modules/ast/AstListener.scala

```diff
@@ -25,6 +25,8 @@ trait AstListener[T] {
   def onRemoveFile(uri: String): Unit
 }
 
+trait WorkspaceContentListener // this just works as a "marker" for both BaseUnitListener and NewConfigurationListener
```

Comment:
why not extend `AstListener[T]`?

Reply:
The original idea was to simplify without extending AstListener but given the final implementation, totally makes sense to extend